File: eecs168/2020fall/MWF/2020.11.18.classes-part-7/Account/Account.cpp
//Account.cpp
#include "Account.h"

Account::Account(double startingBalance)
{
	if(startingBalance > 0)
	{
		balance = startingBalance;
	}
	else
	{
		balance = 0;
	}
}

bool Account::deposit(double amount)
{
	if(amount > 0)
	{	
		balance = balance + amount;
		return (true);
	}	
	else
	{
		return (false);
	}
}

bool Account::withdraw(double amount)
{
	if(amount <= balance && amount > 0)
	{
		balance = balance - amount;
		return (true);
	}
	else
	{
		return(false);
	}
}

double Account::checkBalance() const
{	
	return(balance);
}


bool Account::transfer(Account& destination, double amount)
{
	if(amount > 0 && amount <= balance)
	{
		balance = balance - amount;
		// option: destination.deposit(amount)
		destination.balance += amount;
		return (true);
	}
	else
	{
		return (false);
	}
}
		

File: eecs168/2024spring/TR/2024.04.09.classes-part-3/account.py
#account.py

class Account:
    def __init__(self):
        self._balance = 0

    def deposit(self, amount):
        if amount > 0:
            self._balance += amount
        else:
            pass

    def withdraw(self, amount):
        #define this
        pass

    def transfer(self, other, amount):
        #define this
        pass

    def check_balance(self):
        return self._balance

    def __gt__(self, other):
        if self._balance > other._balance:
            return True
        else:
            return False
